### PR TITLE
Rework parser and writer (G3 prep)

### PR DESCRIPTION
**Pull request by simoncozens**
I'm going to do the Glyphs 3 branch again, now I know what I'm doing.

This PR is preparatory work for Glyphs 3 support, but there is nothing G3 specific about it at this stage. It consists of three commits, each of which passes tests independently when applied in sequence.

1. The first commit replaces the existing parser with one which uses openstep-plist. It has a facility which allows classes to deserialise their properties.
2. The second adds a facility to the writer allowing classes to serialize themselves, and then uses this to write out the data "by hand" instead of using magic properties (_classesForName/shouldWriteValueForKey/_keyOrder). This seems like a backwards step, but it really helps when adding Glyphs 3 support.
3. The third removes those properties which were there to support the automagic serialization.

### CODE REVIEW

**Comment by simoncozens** [comment]
Well, the *first* commit is a bad rebase, but the other three are as described...

File: Lib/glyphsLib/classes.py [hunk 1]
```diff
@@ -1338,25 +1324,15 @@ def plistValue(self):
...
 
-    _classesForName = {
-        "alignment": str,
-        "angle": parse_float_or_int,
-        "locked": bool,
-        "position": Point,
-        "showMeasurement": bool,
-        "filter": str,
-        "name": str,
-    }
```
Comment (khaledhosny) [inline Lib/glyphsLib/classes.py:1359]: Happy to see this stuff gone!

**Comment by jenskutilek** [comment]
I don't think this is G3-specific, but TrueType hints can have 3 values in origin and target. That happens when the hinted point is one that will only be inserted when converting to TrueType, like an inflection point.

Here's an example file that fails when trying to parse the hints: [S.glyphs.zip](https://github.com/googlefonts/glyphsLib/files/6740618/S.glyphs.zip)

**Comment by simoncozens** [comment]
The hints in your file don't parse using the main branch of glyphsLib, even if I convert it to Glyphs 2 file format, so the fact that the hints don't parse is not specific to this PR. I think what you're saying is a duplicate of #446.

**Comment by jenskutilek** [comment]
Right! Yes, they didn’t ever parse. I just thought you might want to fix it for G3 files :)

**Comment by simoncozens** [comment]
Sure. Happy to fix for G2 *and* G3, but I'd rather do things in separate steps to help reviewers; the "rejigging the parser" component is big enough to be a single PR by itself, we can fix unconnected open issues later...

**Pull request merged**